### PR TITLE
Add telemetry for implemented API Gateway features

### PR DESCRIPTION
**Pull request by hunterwerlla**
## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/apigateway/commands/copyUrl.ts [hunk 1]
```diff
@@ -61,6 +63,7 @@ export async function copyUrlCommand(
         window.showInformationMessage(
             localize('AWS.apig.copyUrlNoStages', "Failed to copy URL because '{0}' has no stages", node.name)
         )
+        recordApigatewayCopyUrl({ result: 'Failed' })
```
Comment (rli) [inline src/apigateway/commands/copyUrl.ts:66]: if we get here it's most likely a user-issue, should we still consider this a fail?

Comment (hunterwerlla) [reply src/apigateway/commands/copyUrl.ts:66]: We currently don't disambiguate this anywhere else, and this is a rare instance where it's easy to disambiguate. I think we should leave fail in for now and if we ever disambiguate we can do it then along with everything else.

File: package-lock.json [hunk 2]
```diff
@@ -4555,7 +4555,7 @@
                 },
                 "wrappy": {
                     "version": "1.0.2",
-                    "resolved": false,
+                    "resolved": "",
```
Comment (justinmk3) [inline package-lock.json:4558]: looks like churn, don't actually need to commit this flip-flop

Comment (hunterwerlla) [reply package-lock.json:4558]: Curating what we commit from package-lock.json is a huge pain though, it's all whatever npm decides to do

File: src/apigateway/commands/invokeRemoteRestApi.ts [hunk 3]
```diff
@@ -166,11 +168,21 @@ export function createMessageReceivedFunc({
...
                 outputChannel.appendLine(`There was an error invoking`)
                 outputChannel.appendLine(error.toString())
                 outputChannel.appendLine('')
             } finally {
                 postMessage({ command: 'invokeApiFinished' })
+                let method: string | undefined = undefined
+                // only set method if it is not empty or undefined
+                if (message.selectedMethod) {
+                    method = message.selectedMethod.toUpperCase()
+                }
```
Comment (justinmk3) [inline src/apigateway/commands/invokeRemoteRestApi.ts:181]: could avoid some ceremony, and `let`, by using a ternary, without losing clarity

    const method = message.selectMethod ? message.selectedMethod.toUpperCase() : undefined

File: package-lock.json (hunk 2 above)
Comment (justinmk3) [reply package-lock.json:4558]: yeah, though it's easy for cases like this, where the only meaningful change is 1 line. Up to you though

File: src/apigateway/commands/invokeRemoteRestApi.ts (hunk 3 above)
Comment (awschristou) [reply src/apigateway/commands/invokeRemoteRestApi.ts:181]: Does const method = `message.selectMethod?.toUpperCase()` work?

Comment (hunterwerlla) [reply src/apigateway/commands/invokeRemoteRestApi.ts:181]: Changed

Comment (hunterwerlla) [reply src/apigateway/commands/invokeRemoteRestApi.ts]: `Does const method = message.selectMethod?.toUpperCase() work?` Nope because we want null or empty

**Pull request merged**